2018-01-24-emacs-python/main.Rmd

```diff
@@ -0,0 +1,210 @@
+---
+title: "Usando scripts Python no Emacs"
+author: "Fernando Mayer"
+date: "24 de Janeiro de 2018"
+tags: ["python", "emacs"]
+categories: ["python", "emacs"]
+#output:
+#  html_fragment
+#-- Usar enquanto edita.
+output:
+  html_document:
+    theme: flatly
+---
+
+```{r, include = FALSE}
+# Carrega as configurações.
+source("../config/setup.R")
+```
+
+No início de Janeiro de 2018 iniciamos um grupo de estudos sobre o uso
+do Python para estatística. O objetivo era tentarmos descobrir como
+fazer em Python aquilo que fazemos rotineiramente em R. O resultado foi
+muito interessante e vimos que existe muita similaridade entre as duas
+linguagens. Abaixo está a divulgação que fizemos do grupo de estudos.
+
+<iframe src="https://www.facebook.com/plugins/post.php?href=https%3A%2F%2Fwww.facebook.com%2Fleg.ufpr%2Fposts%2F935006519988723&width=400&show_text=false&height=266&appId" width="400" height="266" style="border:none;overflow:hidden" scrolling="no" frameborder="0" allowTransparency="true"></iframe>
+
+Uma das primeiras coisas que analisamos, logo no início do curso, foram
+as IDEs (*Integrated Development Environment* ou Ambiente de
+Desenvolvimento Integrado) mais utilizadas para se escrever e executar
+códigos em Python. No mundo do R, sabemos que o onipresente
```

Comment:
Apenas minha opinião. O uso do termo "onipresente" dá um destaque talvez superestimado para o RStudio. Eu concordo que é a mais completa e amplamente reconhecida IDE para R da "era pós RStudio", ou seja, novos usuários sequer sabem que se pode editar R fora do RStudio. No entanto, a era "pré RStudio" foi bem plural em termos de editores pois haviam desde IDEs (como Tinn-R) até plugins para editores gerais (Emacs, Vim, Gedit, Geany, Eclipse, etc: veja alguns https://r-dir.com/blog/2013/01/list-of-r-editors.html). Apesar de ser uma excelente IDE, o RStudio não superou com relaxada vantagem os recursos de Editores como o Emacs pois para o programador/desenvolvedor, recursos como alta facilidade em tarefas de REGEX, executar scripts shell, navegar/editar diretórios, editar arquivos remotamente e customizar o editor conforme as próprias necessidades são mais importantes do que ter alguns botões/menus/painéis na IDE que executem/apontem/exibam algo. Obviamente que usuários iniciantes preferem a comodidade desses elementos e podem até mesmo desconhecer ou nunca precisar de recursos mais sofisticados.